backend/app/routes/auth_routes.py

```
@@ -0,0 +1,98 @@
+    if not all(k in data for k in ['username', 'password']):
+        return jsonify({'error': '必須フィールドが不足しています'}), 400
+    
+    # ユーザーを検索
+    user = User.query.filter_by(username=data['username']).first()
+    if not user or not check_password_hash(user.password_hash, data['password']):
+        return jsonify({'error': 'ユーザー名またはパスワードが正しくありません'}), 401
+    
+    # JWTトークンを生成
+    token = jwt.encode({
+        'user_id': user.id,
+        'exp': datetime.datetime.utcnow() + datetime.timedelta(days=1)
```

Ruff (0.8.2)

`datetime.datetime.utcnow()` used

(DTZ003)